domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClientStreamResult.java

```diff
@@ -99,9 +100,10 @@ void fireEndHandler() {
    * @param cause
    */
   void fireExceptionHandler(Throwable cause) {
-    if (exceptionHandler != null) {
+    if (!failed && exceptionHandler != null) {
       exceptionHandler.handle(cause);
     }
+    failed = true;
```

Comment:
`failed = true` should be set before calling exceptionHandler just in case that exceptionHandler tries to call fireHandler or again fireExceptionHandler.

Reply:
ok